ona/tests.py

```diff
@@ -0,0 +1,168 @@
+        userproject = UserProject.objects.create(
+            user=self.normaluser)
+        userproject.projects.add(project3)
+        userproject.save()
+        return project3
+
+    def make_category(self, name="test cat", order=1):
+        category1 = Category.objects.create(
+            name=name,
+            order=order,
+            metadata={}
+        )
+        projectcategory = ProjectCategory.objects.create(
+            project=self.project)
+        projectcategory.categories.add(category1)
+        projectcategory.save()
+        return category1
+
+    def make_location(self, x, y):
+        point_data = Location.objects.create(
+            point="SRID=4326;POINT (%s %s)" % (x, y)
+        )
+        point_data.save()
+        return point_data
+
+    def make_report(self, project, metadata={}):
+        category1 = self.make_category(name="Test Cat 1", order=1)
+        location = self.make_location(18.0000000, -33.0000000)
+        report = Report.objects.create(
+            contact_key="579ed9e9c0554eeca149d7fccd9b54e5",
+            to_addr="+27845001001",
+            location=location,
+            incident_at=datetime.datetime.now(),
+            metadata=metadata
+        )
+        report.save()
+        report.categories.add(category1)
+        project.reports.add(report)
+        return report
+
+    def test_create_report_with_active_ona(self):
+        ona_integration = self.make_ona_integration(self.project)
+        self.make_report(self.project)
+        self.assertEqual(ona_integration.submissions.count(), 1)
+
+    def test_create_report_with_inactive_ona(self):
+        ona_integration = self.make_ona_integration(self.project, False)
+        self.make_report(self.project)
+        self.assertEqual(ona_integration.submissions.count(), 0)
+
+    def test_create_report_with_ona_response(self):
+        ona_integration = self.make_ona_integration(self.project)
+        self.make_report(self.project, {"ona_response": "a response"})
+        self.assertEqual(ona_integration.submissions.count(), 0)
+
+    @responses.activate
+    def test_send_submission_task_success(self):
+        self._restore_post_save_hooks()
+        responses.add(
+            responses.POST,
+            "https://ona.io/api/v1/submissions",
+            body='{"message": "Successful submission"}', status=200,
+            content_type='application/json'
+        )
+
+        report = self.make_report(self.project)
+        ona_integration = self.make_ona_integration(self.project)
+
+        submission = Submission.objects.create(
+            report=report,
+            integration=ona_integration,
+            content="{some content to send}"
+        )
+        submission.save()
+        updated_report = Report.objects.get(pk=report.id)
+        self.assertEqual(
+            updated_report.metadata["ona_response"],
+            "Successful submission"
+        )
+        self._replace_post_save_hooks()
```

Comment:
One slight problem with the `_restore_post_save_hooks / _replace_post_save_hooks` pattern used here is that if the test fails, it will throw an additional error from `tearDown` because `_restore_post_save_hooks` will run again (without `_replace_post_save_hooks` having been re-run).

The most straight forward way to fix this is to do:
```
self._restore_post_save_hooks()
try:
   ... rest of test ...
finally:
    self._replace_post_save_hooks()
```
which will ensure that `self._replace_post_save_hooks()` runs before the function exits.